src/qamqpqueue_p.h

```diff
@@ -19,6 +22,48 @@ class QAmqpQueuePrivate: public QAmqpChannelPrivate,
         METHOD_ID_ENUM(miDelete, 40)
     };
 
+    /*!
+     * A class representing the state of this queue's subscriptions to
+     * a given exchange.
+     */
+    class SubscriptionState {
+    public:
+        /*! Exchange pointer, for checking state */
+        QPointer<QAmqpExchange> exchange;
```

Comment:
hmm I think the need for an exchange inside every single queue is a little suspect. Can we stick to just the QAmqpChannel + QAmqpExchange changes for these commits and then move on to the QAmqpQueue changes? it's going to make review that much easier/faster